src/Core/Domain/Category/Command/AddCategoryCommand.php

```diff
@@ -0,0 +1,87 @@
+<?php
+/**
+ * 2007-2018 PrestaShop.
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Domain\Category\Command;
+
+use PrestaShop\PrestaShop\Core\Domain\Category\Exception\CategoryConstraintException;
+
+/**
+ * Class AddCategoryCommand adds new category.
+ */
+class AddCategoryCommand extends AbstractRootCategoryCommand
+{
+    /**
+     * @var int
+     */
+    private $parentCategoryId;
+
+    /**
+     * @param string[] $name
+     * @param string[] $linkRewrite
+     * @param bool $isActive
+     * @param int $parentCategoryId
+     *
+     * @throws CategoryConstraintException
+     */
+    public function __construct(array $name, array $linkRewrite, $isActive, $parentCategoryId)
```

Comment:
same as previous comments, array $names or string $name? A category can have many names?

Reply:
same as above, it depends on how you think about it, is it single name in different languages or is it different names?

Reply:
In both situation I think if you have a name in different languages, you have many languages, so, many names :/